core/src/com/unciv/logic/automation/civilization/NextTurnAutomation.kt

```diff
@@ -816,19 +871,26 @@ object NextTurnAutomation {
                     && !civInfo.getDiplomacyManager(it).hasFlag(DiplomacyFlags.DeclinedDefensivePact)
                     && civInfo.getDiplomacyManager(it).relationshipIgnoreAfraid() == RelationshipLevel.Ally
             }
-            .sortedByDescending { it.stats.statsForNextTurn.science }
 
         for (otherCiv in canSignDefensivePactCiv) {
-            // Default setting is 1, this will be changed according to different civ.
-            if ((1..10).random() > 1) continue
-            //todo: Add more in depth evaluation here
-            val tradeLogic = TradeLogic(civInfo, otherCiv)
-            tradeLogic.currentTrade.ourOffers.add(TradeOffer(Constants.defensivePact, TradeType.Treaty))
-            tradeLogic.currentTrade.theirOffers.add(TradeOffer(Constants.defensivePact, TradeType.Treaty))
-
-            otherCiv.tradeRequests.add(TradeRequest(civInfo.civName, tradeLogic.currentTrade.reverse()))
+            // Default setting is 3, this will be changed according to different civ.
+            if ((1..10).random() <= 1 && wantsToSignDefensivePact(civInfo, otherCiv)) {
+                //todo: Add more in depth evaluation here
+                val tradeLogic = TradeLogic(civInfo, otherCiv)
+                tradeLogic.currentTrade.ourOffers.add(TradeOffer(Constants.defensivePact, TradeType.Treaty))
+                tradeLogic.currentTrade.theirOffers.add(TradeOffer(Constants.defensivePact, TradeType.Treaty))
+    
+                otherCiv.tradeRequests.add(TradeRequest(civInfo.civName, tradeLogic.currentTrade.reverse()))
+            }
         }
     }
+    
+    fun wantsToSignDefensivePact(civInfo: Civilization, otherCiv: Civilization): Boolean {
+        if (civInfo.getDiplomacyManager(otherCiv).isRelationshipLevelLT(RelationshipLevel.Ally)) return false
+        val friendlyCivs = civInfo.getDiplomacyManager(otherCiv).getCommonKnownCivs().filter { it.getDiplomacyManager(civInfo).isRelationshipLevelGE(RelationshipLevel.Friend) }
```

Comment:
Please change this to "for civ in getcommonk owncivs, if a &b, return false"
This double iteration obfuscates what we're actually trying to find

Reply:
I wasn't sure if I wanted to add more conditions to the defensive pact later. But I think at some point we have to push it out and tweak it later.